File: src/project/project05/f.java
public class Fibonacci {

    public static void main(String[] args) {
        long startTime = System.currentTimeMillis(); // 获取开始时间
        for (int i = 1; i < 50; i++) {
            f(i);  // 调用 Fibonacci 函数
        }
        // 计算并输出程序执行时间
        long duration = System.currentTimeMillis() - startTime;
        System.out.println("Execution time in milliseconds: " + duration);
    }

    // Fibonacci 函数
    public static int f(int n) {
        if (n == 1) {
            return 1;
        } else if (n == 2) {
            return 2;
        } else {
            return f(n - 1) + f(n - 2);  // 递归计算 Fibonacci 数列
        }
    }
}

File: project05/f.go
package main

import(
"fmt"
"time"
)


func main() {
	startTime := time.Now() // 获取开始时间
	for i := 1; i < 50; i++ {
		f(i) 
	}
	// 计算并输出程序执行时间
    duration := time.Since(startTime)
    fmt.Println("Execution time in milliseconds:", duration.Milliseconds())
}

func f(n int) int {
	switch {
	case n == 1:
		return 1
	case n == 2:
		return 2
	default:
		return f(n-1) + f(n-2)
	}
}
